### DISCUSSION TITLE
static and server side noone is not good for page generating

### DISCUSSION THREAD

**Discussion by loveyandex** [vercel/next.js#32851] [Help]
your design is not good for page generation 
if we have a e-commerce website  we can not create new product in runtime 
we should stop nextjs server and rebuild project to add new product.......

we should have realtime static generation worker..

what should i do in production level?
is there solution for it 

### Expected Behavior

latest

### To Reproduce

latest

**Comment by grefrit** [comment]
ISR is super useful when u need to update data periodically. That's probably feature you described 
https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration

**Comment by gitsupi** [comment]
@GrefriT  thanks man for ISR

but we need also a webbook listening for updating page instead revalidating by time scheduler 

for example we send an event (/update/page/blog) to immediately nextjs regenerate statics pages(blog)....
a part from scheduling model...

do you know this possibility exists?

**Comment by grefrit** [comment]
@gitsupi But you can set revalidate for example 10 seconds, yes it won't be instant, but there is no alternatives. Or you can use SSR, but that's will cost you performance. So the best way it just set revalidate low value
